### PR TITLE
Fix PublicSuffixMatcher for wildcard domains

### PR DESCRIPTION
**Pull request by jeromedemangel**

### CODE REVIEW

**Comment by ok2c** [comment]
@jeromedemangel Is this related to [HTTPCLIENT-1997](https://issues.apache.org/jira/browse/HTTPCLIENT-1997)?

**Comment by jeromedemangel** [comment]
> @jeromedemangel Is this related to [HTTPCLIENT-1997](https://issues.apache.org/jira/browse/HTTPCLIENT-1997)?

I think yes

**Comment by ok2c** [comment]
@jeromedemangel Great. The proposed fix looks good to me. Could you please squash all commits into one and make sure all tests pass?

**Comment by ok2c** [comment]
@jeromedemangel Please see my commit b6c9688e21a3b55677913aeacc1eb80aabc86b4b. Please integrate test cases from that commit into your fix.

File: httpclient5/src/test/java/org/apache/hc/client5/http/psl/TestPublicSuffixMatcher.java [hunk 1]
```diff
@@ -70,6 +70,8 @@ public void testGetDomainRoot() throws Exception {
         Assert.assertEquals("blah.ac.jp", matcher.getDomainRoot("blah.blah.ac.jp"));
         Assert.assertEquals("garbage", matcher.getDomainRoot("garbage"));
         Assert.assertEquals("garbage.garbage", matcher.getDomainRoot("garbage.garbage"));
+        Assert.assertEquals("garbage.garbage", matcher.getDomainRoot("garbage.garbage.garbage.garbage"));
+        Assert.assertEquals("garbage.garbage", matcher.getDomainRoot("*.garbage.garbage"));
```
Comment (ok2c) [inline httpclient5/src/test/java/org/apache/hc/client5/http/psl/TestPublicSuffixMatcher.java]: @jeromedemangel I do not think this is correct. The registerable part of `garbage.garbage` should be `garbage`. Please see b6c9688.

**Comment by ok2c** [comment]
@jeromedemangel Fixed by 19a87e0c772558231e571f99e0a05c0f32ff18eb.

**Pull request closed without merging**